pymc/math.py

```diff
@@ -358,6 +358,22 @@ def log1mexp_numpy(x, *, negative_input=False):
     return out
 
 
+def log1pexp_numpy(x):
+    """Return log(1 + exp(x)).
+
+    This function is numerically more stable than the naive approach.
+    """
+    return np.where(
+        x < -37.0,
```

Comment:
where is that algorithm from?

Reply:
Copied from Aesara, there's a reference in the `at.log1pexp`. I will add it.

Also, this should not really be public, I'll take it from all.